pkg/rbac/cache.go

```diff
@@ -43,5 +43,10 @@ var cacheInst = Cache{
 
 // Get a reference to the cache instance.
 func GetCache() *Cache {
+	// Workaround. Update cache connection with every request.
+	// We need a better way to maintain this connection.
+	cacheInst.pool = db.GetConnection()
+	cacheInst.shared.pool = db.GetConnection()
```

Comment:
why do we need to reference two pools in cache?

Reply:
This is because of the change I made in PR #104 
I made the `SharedData` an independent struct with the goal that tests don't need to mock the entire Cache object.
However, the side effect was this duplication of the `pool` object and of course the canary problem.
Let's discuss tomorrow.